### PR TITLE
Added small script to test installation on with different dependencies

### LINKED ISSUE #156: [Symfony 4.4] No PSR-17 response factory found
It seems https://github.com/php-http/discovery/issues/149 is still not fixed. The fix that was proposed in https://github.com/php-http/discovery/pull/150 was reverted in https://github.com/php-http/discovery/pull/151

> No PSR-17 response factory found. Install a package from this list: https://packagist.org/providers/psr/http-factory-implementation

>  Could not find resource using any discovery strategy. Find more information at http://docs.php-http.org/en/latest/discovery.html#common-errors
>  - Puli Factory is not available
>  - No valid candidate found using strategy "Http\Discovery\Strategy\CommonClassesStrategy". We tested the following candidates: .
>  - No valid candidate found using strategy "Http\Discovery\Strategy\CommonPsr17ClassesStrategy". We tested the following candidates: Nyholm\Psr7\Factory\Psr17Factory, Zend\Diactoros\ResponseFactory, Http\Factory\Diactoros\ResponseFa
  ctory, Http\Factory\Guzzle\ResponseFactory, Http\Factory\Slim\ResponseFactory.

### PR DESCRIPTION
**Pull request by Nyholm**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | Fix #156
| Documentation   |
| License         | MIT

This is a small script to make sure we test different sets of configuration. Related to #156 

Test should fail because #156 describes a problem.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +35,38 @@ matrix:
...
+                    (exit $ok) &&
+                        echo -e "\\e[32mOK\\e[0m $title\\n\\ntravis_fold:end:$fold" ||
+                        echo -e "\\e[41mKO\\e[0m $title\\n"
+                    (exit $ok)
+                }
+                export -f install_test
+
+          script:
+              # Test that we find Guzzle
+              - install_test "will-find" "Http\Discovery\HttpClientDiscovery::find();" "php-http/guzzle6-adapter"
```
Comment (dbu) [inline .travis.yml]: quotes are inconsistent with lines below

File: .travis.yml [hunk 2]
```diff
@@ -35,6 +35,38 @@ matrix:
...
+                    echo -e "\\e[1;34m$title\\e[0m"
+                    echo "./tests/install.sh \"$1\" \"$2\" \"$3\""
+                    ./tests/install.sh "$1" "$2" "$3" 2>&1
+                    local ok=$?
+                    (exit $ok) &&
+                        echo -e "\\e[32mOK\\e[0m $title\\n\\ntravis_fold:end:$fold" ||
+                        echo -e "\\e[41mKO\\e[0m $title\\n"
+                    (exit $ok)
+                }
+                export -f install_test
```
Comment (dbu) [inline .travis.yml]: could we add this as a bash file to tests folder, or `.travis` folder? no big deal, but it seems a bit much for the travis configuration file

File: .travis.yml (hunk 1 above)
Comment (Nyholm) [reply .travis.yml]: Thank you

File: .travis.yml (hunk 2 above)
Comment (Nyholm) [reply .travis.yml]: I dont think so. Im not sure how you can "export" a function in bash/travis.

**Comment by Nyholm** [comment]
Sure, I added a commit with a fix for #156

**Comment by Nyholm** [comment]
> can we improve output of the build?

What do you suggest?

- Each test is folded so it get easier to see what is what. 
- Each tests ends with a green "OK" or a red "KO" to signal status. 
- The second last line of the tests prints some output that is helpful.

File: .travis.yml (hunk 2 above)
Comment (dbu) [reply .travis.yml]: ah i see, bash function, not bash script. i guess thats fine then. maybe bash `source`-ing the file would work, but i will merge as is and tag another patch release

**Comment by dbu** [comment]
i somehow expected something *before* the error line, but the last line is enough. as it should normally not fail, it will at least indicate when something is amiss.

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**